BEIMA.Backend/MongoService/Device.cs

```diff
@@ -116,14 +157,18 @@ private void CheckNullArgument(dynamic arg, string name = "")
         /// <exception cref="ArgumentNullException">Throws exception when any of the required fields are null.</exception>
         public BsonDocument GetBsonDocument()
         {
-            CheckNullArgument(DeviceTag, nameof(DeviceTag));
-            CheckNullArgument(Manufacturer, nameof(Manufacturer));
-            CheckNullArgument(ModelNum, nameof(ModelNum));
-            CheckNullArgument(SerialNum, nameof(SerialNum));
-            CheckNullArgument(YearManufactured, nameof(YearManufactured));
-            CheckNullArgument(Notes, nameof(Notes));
-            CheckNullArgument(Location, nameof(Location));
-            CheckNullArgument(LastModified, nameof(LastModified));
+            if(DeviceTag == null 
```

Comment:
Why was the CheckNullArgument() removed? If we just throw a generic ArgumentNullException, then we lose a lot of details in our Exception messages.

Reply:
Added that back in